graph/src/components/transaction_receipt.rs

```diff
@@ -0,0 +1,216 @@
+    #[sql_type = "Nullable<Binary>"]
+    status: Option<Vec<u8>>,
+}
+
+/// Like web3::types::Receipt, but with fewer fields.
+pub struct LightTransactionReceipt {
+    pub transaction_hash: H256,
+    pub transaction_index: U64,
+    pub block_hash: Option<H256>,
+    pub block_number: Option<U64>,
+    pub gas_used: Option<U256>,
+    pub status: Option<U64>,
+}
+
+impl LightTransactionReceipt {
+    /// `0x0` indicates transaction failure, `0x1` indicates transaction success.
+    /// Set for blocks mined after Byzantium hard fork, `None` before.
+    ///
+    /// Relevant EIPs: 609, 658
+    pub fn is_sucessful(&self) -> Option<bool> {
+        self.status.map(|status| !status.is_zero())
+    }
+}
+
+/// Converts Vec<u8> to [u8; N], where N is the vector's expected lenght.
+/// Fails if input size is larger than output size.
+pub(crate) fn drain_vector<const N: usize>(input: Vec<u8>) -> Result<[u8; N], anyhow::Error> {
+    anyhow::ensure!(input.len() <= N, "source is larger than output");
+    let mut output = [0u8; N];
+    let start = output.len() - input.len();
+    output[start..].iter_mut().set_from(input);
+    Ok(output)
+}
+
+#[test]
+fn test_drain_vector() {
+    let input = vec![191, 153, 17];
+    let expected_output = [0, 0, 0, 0, 0, 191, 153, 17];
+    let result = drain_vector(input).expect("failed to drain vector into array");
+    assert_eq!(result, expected_output);
+}
+
+impl TryFrom<RawTransactionReceipt> for LightTransactionReceipt {
+    type Error = anyhow::Error;
+
+    fn try_from(value: RawTransactionReceipt) -> Result<Self, Self::Error> {
+        let RawTransactionReceipt {
+            transaction_hash,
+            transaction_index,
+            block_hash,
+            block_number,
+            gas_used,
+            status,
+        } = value;
+
+        let transaction_hash = drain_vector(transaction_hash)?;
+        let transaction_index = drain_vector(transaction_index)?;
+        let block_hash = block_hash.map(drain_vector).transpose()?;
+        let block_number = block_number.map(drain_vector).transpose()?;
+        let gas_used = gas_used.map(drain_vector).transpose()?;
+        let status = status.map(drain_vector).transpose()?;
+
+        Ok(LightTransactionReceipt {
+            transaction_hash: transaction_hash.into(),
+            transaction_index: transaction_index.into(),
+            block_hash: block_hash.map(Into::into),
+            block_number: block_number.map(Into::into),
+            gas_used: gas_used.map(Into::into),
+            status: status.map(Into::into),
+        })
+    }
+}
+
+/// Queries the database for all the transaction receipts in a given block range.
+pub fn find_transaction_receipts_for_block(
+    conn: &PgConnection,
+    chain_name: &str,
+    block_hash: &H256,
+) -> anyhow::Result<Vec<LightTransactionReceipt>> {
+    let query = TransactionReceiptQuery {
+        schema_name: chain_name,
+        block_hash: block_hash.as_bytes(),
+    };
```

Comment:
I'm still not sure if `chain_name` maps strictly to `schema_name`
I'll have that checked.